client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -194,9 +194,10 @@ private ClientConnection getConnection() {
         }
 
         /**
-         * @param throwable Error that has occurred that needs to be handled by tearing down the connection.
+         * @param ex Error that has occurred that needs to be handled by tearing down the connection.
          */
-        private void failConnection(Throwable throwable) {
+        private void failConnection(Throwable ex) {
+            Throwable throwable = Exceptions.unwrap(ex);
```

Comment:
Currently SegmentOutputStreamImpl.failConnection() is doing this unwrapping before invoking this. If you want the unwrapping here, please remove it from the callers.

Reply:
Thanks, I have removed the redundant `Exceptions.unwrap(ex)`